tests/qmctl-test/scripts/qmctl_json_output.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+
+# Exit immediately on any error
+set -e
+
+# shellcheck disable=SC1091
+. ../e2e/lib/utils
+
+# Install development tools if needed
+exec_cmd "dnf install --setopt=reposdir=/etc/yum.repos.d  --installroot=/usr/lib/qm/rootfs -y hostname ss || true"
+
+QMCTL_SCRIPT="../../tools/qmctl/qmctl"
+
+info_message "Starting qmctl JSON output tests..."
+
+# Test 1: Show commands with JSON output
+run_test "JSON Show container" 0 "json_valid" "" \
+    python3 "$QMCTL_SCRIPT" show container --json
+
+# Handle unix-domain-sockets separately (may have missing ss command)
+info_message "Testing: JSON Show unix-domain-sockets"
+unix_json_file=$(mktemp)
+python3 "$QMCTL_SCRIPT" show unix-domain-sockets --json > "$unix_json_file" 2>&1
+unix_json_exit=$?
+
+if [ "$unix_json_exit" -eq 0 ]; then
+    if python3 -m json.tool < "$unix_json_file" > /dev/null 2>&1; then
+        pass_message "JSON Show unix-domain-sockets - Valid JSON output"
+    else
+        fail_message "JSON Show unix-domain-sockets - Invalid JSON output"
+        exit 1
+    fi
+elif [ "$unix_json_exit" -eq 1 ] && grep -q "ss.*not found\|executable file.*ss.*not found" "$unix_json_file"; then
+    if python3 -m json.tool < "$unix_json_file" > /dev/null 2>&1; then
+        info_message "SKIP: JSON Show unix-domain-sockets - Missing 'ss' command (valid JSON error)"
+    else
+        fail_message "JSON Show unix-domain-sockets - Invalid JSON error response"
+        exit 1
+    fi
+else
+    fail_message "JSON Show unix-domain-sockets - Unexpected error (exit $unix_json_exit)"
+    exit 1
+fi
+rm -f "$unix_json_file"
+
+run_test "JSON Show shared-memory" 0 "json_valid" "" \
+    python3 "$QMCTL_SCRIPT" show shared-memory --json
+
+run_test "JSON Show available-devices" 0 "json_valid" "" \
+    python3 "$QMCTL_SCRIPT" show available-devices --json
+
+run_test "JSON Show namespaces" 0 "json_valid" "" \
+    python3 "$QMCTL_SCRIPT" show namespaces --json
+
+# Test 2: Exec commands with JSON output
+info_message "QM container is available - testing exec JSON commands"
+
+run_test "JSON Exec echo" 0 "json_valid" "" \
+    python3 "$QMCTL_SCRIPT" exec --json echo "hello world"
+
+# Handle hostname separately (may not work in container)
+info_message "Testing: JSON Exec hostname"
+hostname_json_file=$(mktemp)
+python3 "$QMCTL_SCRIPT" exec --json hostname > "$hostname_json_file" 2>&1
+hostname_json_exit=$?
+
+if [ "$hostname_json_exit" -eq 0 ]; then
+    if python3 -m json.tool < "$hostname_json_file" > /dev/null 2>&1; then
+        pass_message "JSON Exec hostname - Valid JSON output"
+    else
+        fail_message "JSON Exec hostname - Invalid JSON output"
```

Comment:
if fail we need to remove hostname_json_file

Reply:
Removed